reflection.md

```diff
@@ -15,4 +40,36 @@ Write a few brief paragraphs that discuss:
    you implement it _or_ what features of Scala would prevent you from
    implementing it? (You don't have to write code for this part. You could say
    "I would use literal extension to..." or "Scala's rules for valid
-   identifiers prevent...")
\ No newline at end of file
+   identifiers prevent...")
+
+As a whole, I think the design is intuitive and not very verbose.  In particular,
+|| seems like a natural union operator because of its other CS definitions.  * Is
+a pretty common "zero or more" operator, and + has an expected behavior from other
+languages.  In addition, the literal extensions work well to save the user from
+concatenating many characters.
+
+On the otherhand, the design differs from other regex languages.  I'm not sure why
+we used the `<*>` operator, instead of just `*`, which is more common.  Same
```

Comment:
I think we used this because `*` and `+` are already defined for Char and String. This means that if we had `"ab"*` I think it would try to do the string `*` operator and expect an integer afterwards. However, because it expects that integer after I'm not certain if it would try char as a regular expression or just give up.